sklearn/metrics/_scorer.py

```diff
@@ -64,20 +65,23 @@
 
 from ..utils.multiclass import type_of_target
 from ..base import is_regressor
+from ..utils._response import _get_response_values
 from ..utils._param_validation import HasMethods, StrOptions, validate_params
 
 
-def _cached_call(cache, estimator, method, *args, **kwargs):
+def _cached_call(cache, estimator, response_method, *args, **kwargs):
     """Call estimator with method and args and kwargs."""
-    if cache is None:
-        return getattr(estimator, method)(*args, **kwargs)
+    if cache is not None and response_method in cache:
```

Comment:
Given the large scope of this PR, can these updates to `_scorer` be made on `main` as an independent PR?

Reply:
It is already: https://github.com/scikit-learn/scikit-learn/pull/26037

Reply:
This has been merged in `main` and is not anymore in the diff.